### DISCUSSION TITLE
Is it possible to have editable text boxes with newlines?

### DISCUSSION THREAD

**Discussion by duarte-pompeu** [zauberzeug/nicegui#381] [Q&A] [answered]
Is this supported?

I tried:
- `nicegui.ui.input` but no luck: editable but I couldn't get newlines.
- `nicegui.ui.log` which allows multi lines, but didn't seem to be editable.

**Comment by falkoschindler** [comment, marked as answer]
Yes, there is. Since `ui.input` uses [Quasar's Input component](https://quasar.dev/vue-components/input), we can use the prop `type=textarea`:

```py
ui.input('Hello world!').props('type=textarea')
```

https://github.com/zauberzeug/nicegui/discussions/27#discussioncomment-2383036

**Reply by duarte-pompeu** [reply]
Thanks! This did it for me:

```python
ui.input().props("type=textarea autogrow filled style=width:500px")
```

Result:

![image](https://user-images.githubusercontent.com/7232812/218880848-e59aee65-586e-445d-820f-8b32e5866bdb.png)
